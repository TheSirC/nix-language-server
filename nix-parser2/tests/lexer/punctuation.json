[
  {
    "kind": "At",
    "span": {
      "start": 0,
      "end": 1
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 1,
      "end": 2
    }
  },
  {
    "kind": "Colon",
    "span": {
      "start": 2,
      "end": 3
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 3,
      "end": 4
    }
  },
  {
    "kind": "Comma",
    "span": {
      "start": 4,
      "end": 5
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 5,
      "end": 6
    }
  },
  {
    "kind": "Dot",
    "span": {
      "start": 6,
      "end": 7
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 7,
      "end": 8
    }
  },
  {
    "kind": "Ellipsis",
    "span": {
      "start": 8,
      "end": 11
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 11,
      "end": 12
    }
  },
  {
    "kind": "Eq",
    "span": {
      "start": 12,
      "end": 13
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 13,
      "end": 14
    }
  },
  {
    "kind": "Interpolate",
    "span": {
      "start": 14,
      "end": 16
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 16,
      "end": 17
    }
  },
  {
    "kind": "OpenBrace",
    "span": {
      "start": 17,
      "end": 18
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 18,
      "end": 19
    }
  },
  {
    "kind": "CloseBrace",
    "span": {
      "start": 19,
      "end": 20
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 20,
      "end": 21
    }
  },
  {
    "kind": "OpenBracket",
    "span": {
      "start": 21,
      "end": 22
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 22,
      "end": 23
    }
  },
  {
    "kind": "CloseBracket",
    "span": {
      "start": 23,
      "end": 24
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 24,
      "end": 25
    }
  },
  {
    "kind": "OpenParen",
    "span": {
      "start": 25,
      "end": 26
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 26,
      "end": 27
    }
  },
  {
    "kind": "CloseParen",
    "span": {
      "start": 27,
      "end": 28
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 28,
      "end": 29
    }
  },
  {
    "kind": "Semi",
    "span": {
      "start": 29,
      "end": 30
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 30,
      "end": 31
    }
  },
  {
    "kind": "Unknown",
    "span": {
      "start": 31,
      "end": 32
    }
  },
  {
    "kind": "Whitespace",
    "span": {
      "start": 32,
      "end": 33
    }
  }
]