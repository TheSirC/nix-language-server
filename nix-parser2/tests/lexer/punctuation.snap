Kind: At, Span: [0, 1), Text: "@"
Kind: Whitespace, Span: [1, 2), Text: "\n"
Kind: Colon, Span: [2, 3), Text: ":"
Kind: Whitespace, Span: [3, 4), Text: "\n"
Kind: Comma, Span: [4, 5), Text: ","
Kind: Whitespace, Span: [5, 6), Text: "\n"
Kind: Dot, Span: [6, 7), Text: "."
Kind: Whitespace, Span: [7, 8), Text: "\n"
Kind: Ellipsis, Span: [8, 11), Text: "..."
Kind: Whitespace, Span: [11, 12), Text: "\n"
Kind: Eq, Span: [12, 13), Text: "="
Kind: Whitespace, Span: [13, 14), Text: "\n"
Kind: Interpolate, Span: [14, 16), Text: "${"
Kind: Whitespace, Span: [16, 17), Text: "\n"
Kind: OpenBrace, Span: [17, 18), Text: "{"
Kind: Whitespace, Span: [18, 19), Text: "\n"
Kind: CloseBrace, Span: [19, 20), Text: "}"
Kind: Whitespace, Span: [20, 21), Text: "\n"
Kind: OpenBracket, Span: [21, 22), Text: "["
Kind: Whitespace, Span: [22, 23), Text: "\n"
Kind: CloseBracket, Span: [23, 24), Text: "]"
Kind: Whitespace, Span: [24, 25), Text: "\n"
Kind: OpenParen, Span: [25, 26), Text: "("
Kind: Whitespace, Span: [26, 27), Text: "\n"
Kind: CloseParen, Span: [27, 28), Text: ")"
Kind: Whitespace, Span: [28, 29), Text: "\n"
Kind: Semi, Span: [29, 30), Text: ";"
Kind: Whitespace, Span: [30, 31), Text: "\n"
Kind: Unknown, Span: [31, 32), Text: "^"
Kind: Whitespace, Span: [32, 33), Text: "\n"